### DISCUSSION TITLE
RS232 Hardware Recommendation

### DISCUSSION THREAD

**Discussion by inzinga** [WillCodeForCats/tekmar-packetserv#47] [Q&A] [answered]
I have Tekmar tn4 controller with Gateway 482 as part of system.  I never connected the 482 to a home automation system.

I'm looking for recommendations related to the last two bullets on the required hardware list of your tekmar-packetserv home assistant plugin. 

- RS232 Serial Port or a USB to Serial adapter
- RS232 Serial to IP converter for Socket or RFC2217

I'd prefer to connect wirelessly from 482 to my home wifi rather than running ethernet from the 482 to router.  Could you provide a couple of specific product references for the above two items?

**Comment by WillCodeForCats** [comment, marked as answer]
You only need one or the other. USB serial adapter if direct connected or the Serial to IP converter if over network.

This one was mentioned in #26. It's wired but I would guess there is probably a wifi version too.
* https://www.amazon.com/gp/product/B07G5P4CPR

Personally I use a Perle IOLAN SDG2 W device server:
* https://store.perle.com/iolansds2-w

For USB serial I use FTDI based adapters like this Startech model:
* https://www.startech.com/en-us/cards-adapters/icusb2321f

Or this one is also confirmed to work (Prolific based):
* https://www.startech.com/en-us/cards-adapters/icusb232pro

**Reply by inzinga** [reply]
Thanks very much.
